crates/wasmi/src/memory/buffer.rs

```diff
@@ -20,33 +27,157 @@ impl ByteBuffer {
     /// - If the initial length is 0.
     /// - If the initial length exceeds the maximum supported limit.
     pub fn new(initial_len: usize) -> Self {
+        let mut vec = vec![0x00_u8; initial_len];
+        let (ptr, len, capacity) = (vec.as_mut_ptr(), vec.len(), vec.capacity());
+        core::mem::forget(vec);
```

Comment:
```suggestion
        let mut vec = ManuallyDrop::new(vec![0x00_u8; initial_len]);
        let (ptr, len, capacity) = (vec.as_mut_ptr(), vec.len(), vec.capacity());
```

Reply:
This is how it is implemented in https://doc.rust-lang.org/src/alloc/vec/mod.rs.html#822 and probably a bit safer. We might need to import `ManuallyDrop` at the top though.